### PR TITLE
Unit test fixes

### PR DESCRIPTION
**Pull request by stevenxxiu**
unit test fixes, including:
- fixed parse doctests by sorting the grammar productions

### CODE REVIEW

File: nltk/test/corpus.doctest [hunk 1]
```diff
@@ -697,11 +697,11 @@ distribution of part-of-speech tags for reduplicated words.
     ...     lexeme = record.find('lx').text
     ...     if re.match(r'(.*)\1$', lexeme):
     ...         redup_pos_freqdist[record.find('ps').text] += 1
-    >>> for item in redup_pos_freqdist.keys():
+    >>> for item in sorted(redup_pos_freqdist.keys()):
```
Comment (kmike) [inline nltk/test/corpus.doctest]: maybe use `.most_common()` method?

Comment (stevenxxiu) [inline nltk/test/corpus.doctest]: `.most_common()` has the problem that some frequencies might be the same (not here though), hence not be deterministic.

Comment (kmike) [inline nltk/test/corpus.doctest]: Yes, that's correct. On the other hand, doctests are not only tests, but also a documentation; `sorted(...)` is a bit less readable, and it makes a worse example - users are more likely to use `.most_common` than `sorted` in their own code.

Comment (stevenxxiu) [inline nltk/test/corpus.doctest]: Makes sense, I've changed it to `most_common()` now.

**Comment by stevenbird** [comment]
Thanks @kmike, @jskda.

**Pull request merged**